hrms/hr/report/shift_attendance/shift_attendance.py

```diff
@@ -0,0 +1,166 @@
+		{
+			"label": _("Employee Name"),
+			"fieldname": "employee_name",
+			"fieldtype": "Data",
+			"width": 150,
+		},
+		{
+			"label": _("Shift"),
+			"fieldname": "shift",
+			"fieldtype": "Link",
+			"options": "Shift Type",
+			"width": 120,
+		},
+		{
+			"label": _("Attendance Date"),
+			"fieldname": "attendance_date",
+			"fieldtype": "Date",
+			"width": 120,
+		},
+		{
+			"label": _("In Time"),
+			"fieldname": "in_time",
+			"fieldtype": "Time",
+			"width": 120,
+		},
+		{
+			"label": _("Out Time"),
+			"fieldname": "out_time",
+			"fieldtype": "Time",
+			"width": 120,
+		},
+		{
+			"label": _("Total Working Hours"),
+			"fieldname": "working_hours",
+			"fieldtype": "Data",
+			"width": 120,
+		},
+		{
+			"label": _("Late Entry"),
+			"fieldname": "late_entry",
+			"fieldtype": "Check",
+			"width": 100,
+		},
+		{
+			"label": _("Early Exit"),
+			"fieldname": "early_exit",
+			"fieldtype": "Check",
+			"width": 100,
+		},
+		{
+			"label": _("Department"),
+			"fieldname": "department",
+			"fieldtype": "Link",
+			"options": "Department",
+			"width": 150,
+		},
+		{
+			"label": _("Company"),
+			"fieldname": "company",
+			"fieldtype": "Link",
+			"options": "Company",
+			"width": 150,
+		},
+		{
+			"label": _("Shift Start Time"),
+			"fieldname": "shift_start",
+			"fieldtype": "Time",
+			"width": 120,
+		},
+		{
+			"label": _("Shift End Time"),
+			"fieldname": "shift_end",
+			"fieldtype": "Time",
+			"width": 120,
+		},
+		{
+			"label": _("Shift Actual Start Time"),
+			"fieldname": "shift_actual_start",
+			"fieldtype": "Time",
+			"width": 120,
+		},
+		{
+			"label": _("Shift Actual End Time"),
+			"fieldname": "shift_actual_end",
+			"fieldtype": "Time",
+			"width": 120,
+		},
+	]
+
+
+def get_data(filters):
+	query = get_query(filters)
+	data = query.run(as_dict=True)
+	data = format_data(data)
+	return data
+
+
+def get_query(filters):
+	attendance = frappe.qb.DocType("Attendance")
+	checkin = frappe.qb.DocType("Employee Checkin")
+	query = (
+		frappe.qb.from_(attendance)
+		.inner_join(checkin)
+		.on(checkin.attendance == attendance.name)
+		.select(
+			attendance.employee,
+			attendance.employee_name,
+			attendance.shift,
+			attendance.attendance_date,
+			attendance.in_time,
+			attendance.out_time,
+			attendance.working_hours,
+			attendance.late_entry,
+			attendance.early_exit,
+			attendance.department,
+			attendance.company,
+			checkin.shift_start,
+			checkin.shift_end,
+			checkin.shift_actual_start,
+			checkin.shift_actual_end,
+		)
+		.groupby(attendance.name)
+	)
+	for filter in filters:
+		if filter == "from_date":
+			query = query.where(attendance.attendance_date >= filters.from_date)
+		elif filter == "to_date":
+			query = query.where(attendance.attendance_date <= filters.to_date)
+		else:
+			query = query.where(attendance[filter] == filters[filter])
+	return query
+
+
+def format_data(data):
+	for d in data:
+		d.in_time = d.in_time.time()
+		d.out_time = d.out_time.time()
```

Comment:
Some shifts have start and end timings on different dates (midnight shifts). We will have to show the entire timestamp in such cases. Other rows can only have the time. Same goes for shift start, end, actual start, actual end